### PR TITLE
style proposal for the code list in the programmer view

### PR DESCRIPTION
**Pull request by etienneCharignon**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,3 +25,4 @@ build/Release
 node_modules
 public/bower_components
 reports
+programmer.css
```
Comment (jonathanperret) [inline .gitignore]: Something generates `programmer.css` at the root??

Comment (etienneCharignon) [reply .gitignore]: No, this file is generated in public/stylesheets/programmer.css

This line in the .gitignore ignores all programmer.css in any place.

Do you think that we should be more specific ?

**Comment by jonathanperret** [comment]
Yes! Ignoring the whole `public/stylesheets` directory is much better, thanks.

I saw a new behavior while trying out this branch: when deleting the current program, the editor goes blank, whereas in the current `master` the editor keeps showing the code for the deleted program (you can add it back, for example). Is this intentional?

**Comment by etienneCharignon** [comment]
Ah yes, your right. I did not see that. It was not intentional. I prefer how it was before.

**Comment by jonathanperret** [comment]
Can we/do we want to do something about this?

image

Otherwise I like the new look!

**Comment by etienneCharignon** [comment]
For that, I wonder if could set a max-width as a percentage of half of the screen width ?

I also found a bad behaviour in the list when you choose a too long name. I'm trying to find a solution for that last one.

**Comment by etienneCharignon** [comment]
delete links are not accessible on Firefox, on mac, when there is a long list of creatures… (because of the scrollbar hiding the buttons).

capture d ecran 2018-06-25 a 13 11 51

**Comment by jonathanperret** [comment]
Good catch about the scrollbar. Here in Chrome, on my Macbook Pro, the list tends to get "stuck" sometimes: after selecting a creature or clicking "Go live", I can't scroll the list using two fingers on the trackpad… then it works again. Seems quite random 😢

**Comment by jonathanperret** [comment]
Oh, and remember to plug in a mouse to test scrolling, this is how it looks in Chrome/macOS with a mouse plugged in:
image

(it looks exactly the same in Firefox)

Notice how the scrollbar does not cover the list area — so we shouldn't have to move the delete buttons to the left if it always worked this way.

Notice also that there is an unwanted horizontal scrollbar.

Oh and with the mouse I can get the annoying "no scrolling" behavior all the time: if I click a creature, then don't move the mouse and just use the wheel, it won't scroll. I have to move the mouse to another element to get it to work.

File: source/stylesheets/programmer.less [hunk 2]
```diff
@@ -259,6 +259,7 @@ input:checked + .slider:before {
 #objects {
   margin: 0;
   overflow: scroll;
+  overflow-x: hidden;
```
Comment (jonathanperret) [inline source/stylesheets/programmer.less]: `overflow` is a shorthand property. `overflow: scroll` sets both `overflow-x` and `overflow-y`, and then we override `overflow-x`.

Better to set `overflow-x: hidden; overflow-y: scroll;` explicitly, don't you think?

Comment (etienneCharignon) [reply source/stylesheets/programmer.less]: oops

**Comment by etienneCharignon** [comment]
I'm confused, are you suggesting that we force the display of the vertical scrollbar all the time or ajuste the position of the icons when the scrollbar is displayed ?

I don't reproduce any trouble with icons when I plug a mouse. Was it a problem you wanted to report ? 

I didn't know about plugging a mouse. I discovered that you can also set a preference at the os level to force scrollbar display any time (a good thing to know for testing).
capture d ecran 2018-06-26 a 20 41 06

**Comment by jonathanperret** [comment]
Yeah, I remembered that preference too. By default it switches when you plug a mouse, that's why I suggested that but of course it's easier to just switch the preference.

I didn't mean to report a problem with the icons, for some reason they broke just as I took the screenshot. 🤷‍♂️ 

The two things I wanted to show on that screenshot are:
 * a "normal" vertical scrollbar does not obscure the icons, which is good;
 * there was an horizontal scrollbar, which you fixed since then.

Sorry for the confusion!

**Comment by jonathanperret** [comment]
I made the list wide again because it looked a bit timid all squeezed to the left. With the lines and highlight you added, I feel it's no longer difficult to match the delete buttons with the correct entries. Also, having them far from the names makes it less likely to delete a creature by mistake.
What do you think?

Oh, and just for fun:
image

**Comment by jonathanperret** [comment]
Looks all good to me now, please use "Squash and merge" whenever you feel like it.

**Comment by etienneCharignon** [comment]
I don't know why but the line highlight is very light (not enough contrast) on my home computer. I wonder if it is not related to the nightshift option). Any way, I will probably merge tomorrow if you don't do it during the night :-)

**Comment by jonathanperret** [comment]
🍾 !
@juliendorra do you like the new look?

**Comment by jonathanperret** [comment]
image

There's a non-clickable area on the right…

**Comment by etienneCharignon** [comment]
Damned !

**Comment by etienneCharignon** [comment]
I commited a fix directly on master…

**Pull request merged**